src/widget/wlabel.cpp

```diff
@@ -95,11 +96,8 @@ void WLabel::setText(const QString& text) {
     if (m_elideMode != Qt::ElideNone) {
         QFontMetrics metrics(font());
         // Measure the text for label width
-        // it turns out, that "-2" is required to make the text actually fit
-        // (Tested on Ubuntu Trusty)
-        // TODO(lp#:1434865): Fix elide width calculation for cases where
-        // this text is next to an expanding widget.
-        QString elidedText = metrics.elidedText(m_longText, m_elideMode, width() - 2);
+        m_widthHint = metrics.size(0, m_longText).width() + frameWidth();
+        QString elidedText = metrics.elidedText(m_longText, m_elideMode, width() - frameWidth());
```

Comment:
```suggestion
        m_widthHint = metrics.size(0, m_longText).width() + 2 * frameWidth();
        QString elidedText = metrics.elidedText(m_longText, m_elideMode, width() - 2 * frameWidth());
```

Reply:
I think that was it.
At least now the size doesn't change anymore...

Though I think we should try to figure why the label shrinks when parameters are swapped.